planning/mission_planner/src/mission_planner_lanelet2/mission_planner_lanelet2.cpp

```diff
@@ -295,6 +320,7 @@ autoware_auto_planning_msgs::msg::HADMapRoute MissionPlannerLanelet2::planRoute(
   route_msg.segments = route_sections;
   route_msg.goal_pose = goal_pose_.pose;
```

Comment:
I personally feel uncomfortable modifying isGoalValid() function to get the goal height because that wasn't the function was made for.

 Rather than that I suggest to create a function that takes GoalLaneId to refine the goal height (e.g., refinGoalHeight) and call it before this line. 

example:
```
void refineGoalHeight(const RouteSections &route_sections, &goal_pose, lanelet_map_ptr_){
   // get goal lanelet id
   const auto goal_lane_id = route_sections.back().preferred_primitive_id;
   lanelet::Lanelet goal_lanelet = lanelet_map_ptr_->laneletLayer.get(goal_lane_id);
   projectGoalToMap(goal_lanelet, goal_pose);
}
```